datasets/samplemotordatalimerick.py

```diff
@@ -8,269 +8,82 @@
-
-        return file_cnn_signals
-
-    @staticmethod
-    def create_common_empty_df():
-        """
-        Create empty dataframe
-        """
-        df = pd.DataFrame(columns=SampleMotorDataLimerick.common_dataframe_columns)
-        return df
-
-    @staticmethod
-    def parse_ADXL356C_and_return_common_df_row(file_full_path):
-        """
-        Columns added just for readability can return raw data np array as well,
-        can also add file identifier
-        """
-        df_raw = pd.read_csv(file_full_path, sep=';', header=None)
-        df_raw.rename(
-            columns={0: 'Time', 1: 'Voltage_x', 2: 'Voltage_y',
-                     3: 'Voltage_z', 4: 'x', 5: 'y', 6: 'z'},
-            inplace=True
-            )
-        ss_vibr_x1 = df_raw.iloc[0]['x']
-        ss_vibr_y1 = df_raw.iloc[0]['y']
-        ss_vibr_z1 = df_raw.iloc[0]['z']
-        df_raw["Acceleration_x (g)"] = 50 * (df_raw["Voltage_x"] - ss_vibr_x1)
-        df_raw["Acceleration_y (g)"] = 50 * (df_raw["Voltage_y"] - ss_vibr_y1)
-        df_raw["Acceleration_z (g)"] = 50 * (df_raw["Voltage_z"] - ss_vibr_z1)
-
-        raw_data = df_raw[["Acceleration_x (g)", "Acceleration_y (g)", "Acceleration_z (g)"]]
-        raw_data = raw_data.to_numpy()
-        return ['ADXL356C', os.path.basename(file_full_path).split('/')[-1], raw_data]
-
-    def __makedir_exist_ok(self, dirpath):
-        try:
-            os.makedirs(dirpath)
-        except OSError as e:
-            if e.errno == errno.EEXIST:
-                pass
-            else:
-                raise
-
-    def __init__(self, root, d_type, transform=None,
+    def __init__(self, root, d_type,
+                 transform,
+                 target_sampling_rate_Hz,
+                 signal_duration_in_sec,
+                 overlap_ratio,
+                 eval_mode,
+                 label_as_signal,
+                 random_or_speed_split,
+                 num_end_zeros=num_end_zeros,
+                 num_start_zeros=num_start_zeros,
+                 train_ratio=train_ratio,
+                 accel_in_second_dim=True,
                  download=True,
-                 downsampling_ratio=2,
-                 signal_duration_in_sec=0.25,
-                 overlap_ratio=0.75,
-                 eval_mode=False,
-                 label_as_signal=True,
-                 random_or_speed_split=True,
-                 accel_in_second_dim=True):
-
-        if d_type not in ('test', 'train'):
-            raise ValueError(
-                "d_type can only be set to 'test' or 'train'"
-                )
-
-        if not isinstance(downsampling_ratio, int) or downsampling_ratio < 1:
-            raise ValueError(
-                "downsampling_ratio can only be set to an integer value greater than 0"
-                )
-
-        self.selected_sensor_sr = SampleMotorDataLimerick.sensor_sr_Hz
+                 healthy_file_identifier=healthy_file_identifier,
+                 cnn_1dinput_len=256):
```

Comment:
Please add this parameter to get_datasets as well, otherwise default is always is used and that makes 256

Reply:
Updated